sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JdbcUtils.scala

```diff
@@ -125,8 +129,14 @@ object JdbcUtils extends Logging {
       dialect: JdbcDialect): Iterator[Byte] = {
     val conn = getConnection()
     var committed = false
+    val supportsTransactions = Try( 
```

Comment:
This ends up just swallowing the exception, but does produce the desired effect. The style's off here so would need a change anyway, so how about going back to a simple `try-catch`:

```
val supportsTransactions = try {
  conn.getMetaData()...
} catch {
  case _: Exception => true
}
```